src/bsp/src/posix/src/UserInterface.cpp

```diff
@@ -53,3 +82,50 @@ int UserInterface::printLine(const char* format, va_list args) {
 
     return retValue;
 }
+
+void UserInterface::setRGBLed(RgbColor color) { m_uiState.m_rgbLed = color; }
+
+void UserInterface::setLed(LED led, bool state) {
+    m_uiState.m_ledStates[static_cast<uint>(led)] = state;
+}
+
+void UserInterface::setHexDisplay(uint8_t value) { m_uiState.m_hexDisplay = value; }
+
+void UserInterface::setButtonCallback(Button button,
+                                      buttonCallbackFunction_t callback,
+                                      void* context) {
+    ros::NodeHandle nodeHandle = ros::NodeHandle("~");
+    uint buttonId = static_cast<uint>(button);
+
+    std::string buttonTopic =
+        nodeHandle.param("buttonTopic", std::string("/agent_1/user_interface/button"));
```

Comment:
Shouldn't this be based on the agent UUID?

Reply:
I've matched what we done in the HiveMindBridge ros node in SwarmUS ROS [here](
https://github.com/SwarmUS/SwarmUS-ROS/blob/4d047e4b6a86e77aa659400d2cbb6df7cd5941f9/src/hive_mind_bridge/src/main.cpp#L78)

Reply:
Probably a bug there too.
Because right now you are subscribing to the same topic for every running HM which is problematic

Reply:
The second argument is just the default argument if the param is not available in the node handle, it's possible to send a different param to each nodehandle. That's what we do in the launch files